src/components/RegionSettings.tsx

```diff
@@ -6,33 +6,44 @@ import Button from "@material-ui/core/Button";
 import Card from "@material-ui/core/Card";
 import CardActions from "@material-ui/core/CardActions";
 import CardContent from "@material-ui/core/CardContent";
-import RadioButtonCheckedIcon from "@material-ui/icons/RadioButtonChecked";
-import RadioButtonUncheckedIcon from "@material-ui/icons/RadioButtonUnchecked";
 import MenuItem from "@material-ui/core/MenuItem";
 import FormControl from "@material-ui/core/FormControl";
 import Select from "@material-ui/core/Select";
-import ContinuousSlider from "./Slider";
+import { Slider } from "@material-ui/core";
+import ColorPicker from "./ColorPicker";
+import { RGBColor } from "react-color";
 
-export default function RegionSettings() {
-  const [regions, setRegions] = React.useState<
-    Array<{
-      id: string;
-      type: string;
-      transparency: number;
-      activated: boolean;
-    }>
-  >([]);
+interface Region {
+  id: string;
+  type: string;
+  transparency: number;
+  color: RGBColor;
+  activated: boolean;
+}
+
+interface RegionSettingsProps {
+  regions: Array<Region>;
+  selectableRegions: Array<string>;
+}
 
+export default function RegionSettings(props: RegionSettingsProps) {
+  const [regions, setRegions] = React.useState<Array<Region>>(props.regions);
   const addNewRegion = (e: MouseEvent<HTMLButtonElement>) => {
     setRegions([
       ...regions,
-      { type: "None", transparency: 50, activated: false, id: uuidv4() },
+      {
+        type: "",
+        transparency: 50,
+        color: { r: 225, g: 68, b: 13, a: 1 },
+        activated: false,
+        id: uuidv4(),
+      },
     ]);
   };
 
   const activateNewRegion = (id: string) => {
     const regionToActivate = regions.filter(
-      (r) => r.id === id && r.type !== "None"
+      (r) => r.id === id && r.type !== ""
```

Comment:
Czemu to `None` już nie jest potrzebne?

Reply:
Wole opcje z pustym polem, ale mogę przywrócić None.